docs/lang/static.md

```diff
@@ -1,16 +1,84 @@
-However, the generated hardware to schedule the execution may not be efficient–especially if the program can take advantage of the *latency* information.
+By default, Calyx programs use a *latency-insensitive*, or *dynamic*, model of computation.
+This means that the compiler does not know, track, or guarantee the number of cycles it takes to perform a computation or run a control operator.
+This is in contrast to a *latency-sensitive*, or *static*, model of computation, where the number of cycles a component needs is known to, and honored by, the compiler.
 
-More crucially, however, it is impossible for *latency-insensitive* programs to interact with *latency-sensitive* hardware implemented in RTL.
+In general, latency-insensitivity makes it easier to compose programs.
+It grants the compiler freedom to schedule operators however it wants, as long as it meets the program's dataflow constraints.
 
-Calyx uses the `@static` attribute to provide latency information to various constructs and provides strong guarantees about the generated programs.
+However, there are two drawbacks to this approach.
+First, the generated hardware may not be efficient: if the compiler does not know how long computations take, it must schedule them conservatively.
+Second, it is impossible for *latency-insensitive* programs to interact with *latency-sensitive* hardware implemented in RTL;
+this prevents the use of black-box hardware designs.
+
+To address these issues, Calyx provides a `static` keyword that modifies components and groups, along with static variants of other control operators.
+
+## Static support in the Calyx IL
+
+### Static components
+
+Say we have a multiplier component, `std_mult`, which multiplies the values `left` and `right` and puts the result in `out`.
+Its latency is 3 cycles.
+We can declare it as follows:
+```
+static<3> primitive std_mult[W](go: 1, left: W, right: W) -> (out: W);
+```
+Compare this to the divider component `std_div`, whose latency is unknown:
+```
+primitive std_div[W](go: 1, left: W, right: W) -> (out: W, done: 1)
+```
+The key differences are:
+- The `static` keyword is used to declare the component as static and to specify its latency.
+- The `done` port is not present in the static component.
+
+A client of the divider must pass two inputs, raise the `go` signal, and wait for the component itself to raise its `done` signal.
+In contrast, a client of the multiplier must pass two inputs and raise the `go` signal, but it does not need to wait for the component to raise a `done` signal.
+It can simply and safely assume that the result will be available after 3 cycles.
+
+
+### Static groups and relative timing guards
+
+Much like components, groups can be declared as static.
+Since groups are just unordered sets of assignments, it pays to have a little more control over the scheduling of the assignments within a group.
```

Comment:
Maybe we should explicitly indicate that we are introducing a new feature. Something like: "Static groups also have a unique feature that ordinary dynamic groups do not have: ..."